### ISSUE TITLE
End to end test add to pull request checks

### ISSUE THREAD

**Issue by Sriep** [0chain/0chain#350]
I have fixed a lot of bugs in `0chain` smart contracts where the **numbers** produced were wrong. The fact that these bugs were allowed to happen in the first place is a problem. I suggest adding `end to end` tests to `pull requests check actions` as a solution.

By `end to end` test I mean a set of transaction that form a simple used case, with the enthesis is on confirming the correct numbers are returned. For example:
1. User creates a wallet and gets tokens.
2. Lock tokens.
3. Create an allocation.
4. Save a file. Read a file.
5. Close allocation.
6. Check balances of all token pools; user, blobbers, stakeholders of blobbers, sharders and miners, and anyone else.

So we would need a Dockerfile that fires out a basic 1 sharder, 4 miners, 4 bobbers 0chain network and runs a series of scripts. This can be added to the GitHub actions. It might be most efficient to write these scripts using `gosdk`, but could use `bash` and `curl`. 

Each time a new feature is added, an end to end script can be written to prove the changes. This can be run with pull requests to prevent any regression.

**Comment by guruhubb** [comment]
@bbist I believe the conductor already does this. Can we have it in ci/cd as was discussed a while back.

**Comment by bbist** [comment]
> @bbist I believe the conductor already does this, but not sure if it to the extent @Sriep is talking about, but we can certainly enhance it. Can we have it in ci/cd as was discussed a while back.

Yes, conductor framework, as it is, can be used to spin up necessary nodes as well as test containers in a sequence necessary for the test.
But the this kind of test can just easily be implemented using docker-compose. In fact, it may be more preferable than using conductor framework.

**Comment by Sriep** [comment]
If it is being run on each pull request, then it needs to be efficient. If it can be kept within 10 minutes then it will not increase time for checks as they are done in parallel and currently take 10 minutes. 

Can conductor be fed a sequence of simple scripts and the return results checked without unnecessary baggage?

**Comment by Sriep** [comment]
```
I believe the conductor already does this,
```
`Conductor` would not be suitable as it uses a different code stream to 0chain, so it is not a true 0chain integration test.

**Comment by stewartie4** [comment]
completely agree with this, this is what we want to have with the system tests initiative 
For now we have API end to end tests that can be kicked off manually
https://github.com/0chain/0chain/actions/workflows/manual-api-regressions.yml

**Comment by Sriep** [comment]
Moved to https://github.com/0chain/system_test
